stories/organisms/SelectedItemBadge.stories.tsx

```diff
@@ -161,59 +211,58 @@ export const _SelectedItemBadgeMultiSelect: Story = {
         };
 
         return (
-            <>
-                <Text>
-                    Note: group selection and Select all is not correctly implemented in this
-                    example. We only showcasing the layout here.
-                </Text>
-                <div style={{ width: '200px' }}>
-                    <SelectedItemBadge
-                        {...args}
-                        priority={
-                            args.priority && {
-                                ...args.priority,
-                                selectedItem: selectedPriorityItem,
-                                onChange: handlePriorityChange,
-                            }
-                        }
-                        onDelete={handleDelete}
-                        additionalLabel={
-                            selectedSynonyms.length > 0 && `+${selectedSynonyms.length}`
+            <div style={{ width: '200px' }}>
+                <SelectedItemBadge
+                    {...args}
+                    priority={
+                        args.priority && {
+                            ...args.priority,
+                            selectedItem: selectedPriorityItem,
+                            onChange: handlePriorityChange,
                         }
+                    }
+                    onDelete={handleDelete}
+                    additionalLabel={
+                        selectedSynonyms.length > 0 &&
+                        `+${selectedSynonyms.filter((s) => s !== 'All' && s !== 'English').length}`
+                    }
+                >
+                    <MultiSelectItem
+                        variant="select-all"
+                        isSelected={selectedSynonyms.includes('All')}
+                        onSelect={(event) => handleOnChange(event, 'All')}
```

Comment:
`onSelect` closes the dropdown after a single selection, so multi-selection doesn’t work as expected. Also, in `Dropdown` we use `onCheckedChange`, which supports multi-selection, so it’s now unclear which one to use  🤔

Reply:
onCheckedChange, makes more sense in this case IMO